### PR TITLE
Add support for wallet_switchEthereumChain

### LINKED ISSUE #2944: switch_Ethereum chain doesn't exist Error
### **Description**
Add basic `switchEthereumChain` feature for Dapps to prompt and suggest a user to switch to a specified chain that they already have added to MetaMask. This method will work with default networks like mainnet.

### **Technical Requirements**
- Implement the `wallet_switchEthereumChain` RPC Method (EIP3326)
- Implement the UI that allows the user to accept or reject the request
- Figma - https://www.figma.com/file/dzapXt5s0pKBxb9lZkIoDM/%231352---Confirm-transactions?node-id=3474%3A3906
- Implement warnings in case the parameters are malformed or the chainId is invalid

### **Acceptance Criteria**
- It should successfully switch networks in the case the request is accepted
- It should throw an error for malformed requests or invalid chainId

### **References**
- Sentry issue: https://sentry.io/organizations/metamask/issues/2396504126/?environment=production&project=2299799&query=is%3Aunresolved&sort=user&statsPeriod=14d
- Related to custom network parity issue https://www.notion.so/Allow-me-to-easily-safely-switch-from-a-custom-network-to-mainnet-default-testnets-via-Network--4b8d0f7dce9e4446a111a4146ad6315e
- Technical guidance from Brad: https://consensys.slack.com/archives/G1L7H42BT/p1624408934213000
- Merged PR on extension - https://github.com/MetaMask/metamask-extension/pull/10905
- Proposal in EIP 3326 - https://github.com/rekmarks/EIPs/blob/3326-create/EIPS/eip-3326.md
eip-3326.md

### **Products POV**
* really high on Sentry, 
* couple GH issues opened by devs, 
* extension parity issue. 
* Fixed on extension, looks like lower effort than transaction state (confirm with engineers?)
** can't answer this yet until grooming/specing out work is done

### PR DESCRIPTION
**Pull request by andrepimenta**
**Description**

Adds support for [`wallet_switchEthereumChain`](https://docs.metamask.io/guide/rpc-api.html#other-rpc-methods)

We can use https://andrepimenta.github.io/wallet_switchEthereumChain.html to test.

**Issue**

Resolves #2944

### CODE REVIEW

**Review by ibrahimtaveras00 — CHANGES_REQUESTED** [review]
Works as expected.  
However, there is one case I'm curious about:
- go to https://andrepimenta.github.io/wallet_switchEthereumChain.html
- don't add xDai chain yet, but tap on "Switch xDai chain" 

This error shows up = https://recordit.co/aqfCj5bPOa
Screen Shot 2021-08-30 at 5 01 19 PM

Should we see some type of message appear in this case?

**Review by ibrahimtaveras00 — APPROVED** [review]
We're all good here, 👍🏽

File: app/core/RPCMethods/wallet_switchEthereumChain.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
+import Engine from '../Engine';
+import { ethErrors } from 'eth-json-rpc-errors';
+import { getDefaultNetworkByChainId, isPrefixedFormattedHexString, isSafeChainId } from '../../util/networks';
+import AnalyticsV2 from '../../util/analyticsV2';
+
+const wallet_addEthereumChain = async ({
```
Comment (sethkfman) [inline app/core/RPCMethods/wallet_switchEthereumChain.js:6]: @rickycodes I would think this should be `wallet_switchEthereumChain`? WDYT

File: app/core/RPCMethods/wallet_switchEthereumChain.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+	const { PreferencesController, CurrencyRateController, NetworkController } = Engine.context;
+
+	if (!req.params?.[0] || typeof req.params[0] !== 'object') {
+		throw ethErrors.rpc.invalidParams({
+			message: `Expected single, object parameter. Received:\n${JSON.stringify(req.params)}`
+		});
+	}
+
+	const params = req.params[0];
```
Comment (rickycodes) [inline app/core/RPCMethods/wallet_switchEthereumChain.js:25]:
```suggestion
	const params = req.params?.[0];

	if (!params || typeof params !== 'object') {
		throw ethErrors.rpc.invalidParams({
			message: `Expected single, object parameter. Received:\n${JSON.stringify(params)}`
		});
	}
```

Comment (rickycodes) [reply app/core/RPCMethods/wallet_switchEthereumChain.js:25]: you'll have to verify, but I think this is functionality equivalent.

**Review by rickycodes — APPROVED** [review]
LGTM

**Pull request merged**